org-roam.el

```diff
@@ -529,21 +528,19 @@ Valid states are 'visible, 'exists and 'none."
   "Disable org-roam updating for file.
 
 1. Remove hooks for updating the cache, and the org-roam buffer."
-  (remove-hook 'post-command-hook #'org-roam--maybe-update-buffer)
-  (remove-hook 'after-save-hook #'org-roam--update-cache))
+  (remove-hook 'post-command-hook #'org-roam--maybe-update-buffer t)
+  (remove-hook 'after-save-hook #'org-roam--update-cache t))
 
 (cl-defun org-roam--maybe-update-buffer (&key redisplay)
   "Update `org-roam-buffer' with the necessary information.
 This needs to be quick/infrequent, because this is run at
 `post-command-hook'."
-  (with-current-buffer (window-buffer)
-    (when (and (get-buffer org-roam-buffer)
-               (buffer-file-name (current-buffer))
-               (file-exists-p (file-truename (buffer-file-name (current-buffer))))
-               (or redisplay
-                   (not (string= org-roam-current-file
-                                 (file-truename (buffer-file-name (current-buffer)))))))
-      (org-roam-update (file-truename (buffer-file-name (window-buffer)))))))
+  (let ((buffer (window-buffer)))
+    (when (or redisplay
+              (not (eq org-roam--current-buffer buffer)))
+      (setq org-roam--current-buffer buffer)
+      (org-roam-update (expand-file-name
+                        (buffer-local-value 'buffer-file-truename buffer))))))
```

Comment:
does this not break if the buffer doesn't belong to a file? I think `org-roam--maybe-enable` needs to implement this check then.